cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/reserve_transfer.rs

```diff
@@ -1132,8 +1145,666 @@ fn reserve_transfer_native_asset_from_para_to_para_through_relay() {
+		args: TestArgs::new_para(
+			penpal_location,
+			receiver.clone(),
+			asset_amount_to_send,
+			assets,
+			None,
+			0,
+		),
+	};
+	let mut test = SystemParaToParaTest::new(test_args);
+
+	// The sender needs some Relay tokens to pay for delivery fees locally.
+	// TODO: Also allow payment with different assets locally, right now only works remotely.
+	// <AssetHubWestend as TestExt>::execute_with(|| {
+	// 	assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::Balances::force_set_balance(
+	// 		<AssetHubWestend as Chain>::RuntimeOrigin::root(),
+	// 		sender.clone().into(),
+	// 		ASSET_HUB_WESTEND_ED,
+	// 	));
+	// });
+
+	let sender_initial_balance = AssetHubWestend::execute_with(|| {
+		type Assets = <AssetHubWestend as AssetHubWestendPallet>::Assets;
+		<Assets as Inspect<_>>::balance(asset_id, &sender)
+	});
+	let sender_initial_native_balance = AssetHubWestend::execute_with(|| {
+		type Balances = <AssetHubWestend as AssetHubWestendPallet>::Balances;
+		Balances::free_balance(&sender)
+	});
+	let receiver_initial_balance = PenpalA::execute_with(|| {
+		type ForeignAssets = <PenpalA as PenpalAPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(custom_asset_penpal_pov.clone(), &receiver)
+	});
+
+	// test.set_assertion::<AssetHubWestend>(system_para_to_para_sender_assertions);
+	// test.set_assertion::<PenpalA>(system_para_to_para_receiver_assertions);
+	test.set_dispatchable::<AssetHubWestend>(system_para_to_para_reserve_transfer_assets);
+	test.assert();
+
+	let sender_after_balance = AssetHubWestend::execute_with(|| {
+		type Assets = <AssetHubWestend as AssetHubWestendPallet>::Assets;
+		<Assets as Inspect<_>>::balance(asset_id, &sender)
+	});
+	let sender_after_native_balance = AssetHubWestend::execute_with(|| {
+		type Balances = <AssetHubWestend as AssetHubWestendPallet>::Balances;
+		Balances::free_balance(&sender)
+	});
+	let receiver_after_balance = PenpalA::execute_with(|| {
+		type ForeignAssets = <PenpalA as PenpalAPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(custom_asset_penpal_pov, &receiver)
+	});
+
+	// TODO: When we allow payment with different assets locally, this should be the same, since
+	// they aren't used for fees.
+	assert!(sender_after_native_balance < sender_initial_native_balance);
+	// Sender account's balance decreases.
+	assert_eq!(sender_after_balance, sender_initial_balance - asset_amount_to_send);
+	// Receiver account's balance increases.
+	assert!(receiver_after_balance > receiver_initial_balance);
+	assert!(receiver_after_balance < receiver_initial_balance + asset_amount_to_send);
+}
+
+// ==========================================================================
+// ===== Reserve Transfers - Pool Asset - Parachain->AssetHub->Parachain ====
+// ==========================================================================
+// From Penpal A to Penpal B with AssetHub as the reserve.
+#[test]
+fn reserve_transfer_pool_assets_from_para_to_para_through_asset_hub() {
```

Comment:
Very nice! 💪 

docs suggestion:
```suggestion
// ==========================================================================
// == Reserve Transfers USDT - Parachain->AssetHub->Parachain - pay fees with USDT (asset conversion) ==
// ==========================================================================
// Transfer USDT From Penpal A to Penpal B with AssetHub as the reserve, while paying fees using
// USDT by making use of existing USDT pools on AssetHub and destination.
#[test]
fn reserve_transfer_usdt_from_para_to_para_through_asset_hub() {
```